src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj

```diff
@@ -4,6 +4,8 @@
     <TargetFrameworks>$(NetFrameworkMinimum);netstandard2.0;net7.0</TargetFrameworks>
     <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyVersion>6.0.0.0</AssemblyVersion> <!-- NO-INCREMENT: This version is frozen for .NET Standard and .NETCoreApp because the assembly ships inbox. -->
+    <CLSCompliant>false</CLSCompliant>
+    <IsTrimmable Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetMinimum)'))">true</IsTrimmable>
```

Comment:
What was the reason for this addition?

Reply:
In the old DLLs for S.R.CS.U., this metadata line is there in all 3 versions, although this is not set explicitly [in the release/6.0 ilproj of this assembly](https://github.com/dotnet/runtime/blob/release/6.0/src/libraries/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj).


net6.0:
![image](https://github.com/user-attachments/assets/76582d05-4a49-43d7-a09d-746d2c1a11b3)

net462:
![image](https://github.com/user-attachments/assets/f787b066-fbda-4608-9bb5-e088cb223fcd)

netstandard2.0:
![image](https://github.com/user-attachments/assets/3bce4805-e615-499a-a49f-9c58da83b47b)

I was not seeing this metadata attribute added to the _new_ DLLs, so I decided to set this property in the ilproj, which automatically added the attribute in the metadata.

But now that I learned that it's two separate things to add a property to a proj and to add an assembly metadata, I am thinking that I might have to only add the assembly metadata directly.

I see a bug in this code too: The condition is unnecessary. I need to remove it (if I decide to keep the property):

```suggestion
    <IsTrimmable>true</IsTrimmable>
```

Reply:
I built the release/6.0 branch to see how S.R.CS.Unsafe was getting the `IsTrimmable` assembly metadata element added. I see that it is never added as a property. Instead, it is added directly as AssemblyMetadata in https://github.com/dotnet/runtime/blob/4b4a745fc0152b887eec20da906b98a229973313/eng/versioning.targets#L23-L25
```xml
    <AssemblyMetadata Include="IsTrimmable" Condition="'$(SetIsTrimmable)' != 'false'">
      <Value>True</Value>
    </AssemblyMetadata>
```

So it's not controlled by setting `IsTrimmable` in the project. Instead, it's controlled by _not_ `SetIsTrimmable` to false, then the metadata is added.

I discovered that the majority of the maintenance-packages assemblies do _not_ expect to have this metadata (see [this comment](https://github.com/dotnet/maintenance-packages/pull/135#issuecomment-2418190786)), only two of them need it: S.R.CS.Unsafe and M.IO.Redist.

The surprising thing is that in maintenance-packages, by setting `IsTrimmable` to `true`, the metadata is getting added automatically thanks to [Microsoft.NET.ILLink.targets#L23-L28](https://github.com/dotnet/runtime/blob/ffc5e169be0cd481ea623b11e4d1378fd8526d09/src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets#L23-L28) but I don't understand why we have that different behavior here compared to runtime:

```xml
  <ItemGroup>
    <AssemblyAttribute Condition="'$(IsTrimmable)' == 'true'" Include="System.Reflection.AssemblyMetadata">
      <_Parameter1>IsTrimmable</_Parameter1>
      <_Parameter2>True</_Parameter2>
    </AssemblyAttribute>
  </ItemGroup>
```

Reply:
But I want to be cautious. I don't know if setting the `IsTrimmable` property to true in the proj will cause any unintended consequences. Since runtime is _only_ adding the assembly metadata, but not the `IsTrimmable` property, I'll do exactly the same for these two assemblies.

Reply:
The only reason why the runtime repo sets the attribute instead of just letting SDK set it based on IsTrimmable is that the code pre-dates the property (attribute: https://github.com/dotnet/runtime/pull/48428, property: https://github.com/dotnet/sdk/pull/17607).

IsTrimmable property enables the trim analyzer in addition to adding the attribute. I'm pretty sure we run the analyzer in the dotnet/runtime libraries. Enabling analyzer has no effect on ILProj because it's not Roslyn-based.